sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -912,4 +912,23 @@ abstract class SchemaPruningSuite
       }
     }
   }
+
+  test("SPARK-37577: Fix ClassCastException: ArrayType cannot be cast to StructType") {
+    import testImplicits._
+
+    val schema = StructType(Seq(
+      StructField("o", ArrayType(StructType(
+        Seq(StructField("s", StringType, false),
+          StructField("b", ArrayType(StructType(
+            Seq(StructField("e", StringType, false))), true), false)
```

Comment:
Can we use more intrusive field names? `o`, `s`, `b`, `e` looks a little confusing.

Reply:
sure, i will update it.